runtime/nvqir/NVQIR.cpp

```diff
@@ -569,6 +569,48 @@ void __quantum__qis__exp_pauli(double theta, Array *qubits, char *pauliWord) {
 
 void __quantum__rt__result_record_output(Result *, int8_t *) {}
 
+void __quantum__qis__custom_unitary(std::complex<double> *unitary,
+                                    Array *controls, Array *targets) {
+  auto ctrlsVec = arrayToVectorSizeT(controls);
+  auto tgtsVec = arrayToVectorSizeT(targets);
+  auto numQubits = tgtsVec.size();
+  auto nToPowTwo = (1ULL << numQubits);
```

Comment:
It would probably be good to throw a `std::invalid_argument()` exception if `numQubits >= 64' to make sure this doesn't overflow. It might save some debugging time in the future if the memory ever gets corrupted.

Reply:
Addressed in b3678f7324f795a6f48455904c5789c27033b255